File: sdk/ruby/lib/experimently.rb
require_relative 'experimently/version'
require_relative 'experimently/errors'
require_relative 'experimently/config'
require_relative 'experimently/types'
require_relative 'experimently/evaluator'
require_relative 'experimently/cache'
require_relative 'experimently/http_client'
require_relative 'experimently/client'

# Experimently Ruby SDK
#
# Provides experiment assignment, feature flag evaluation, and event tracking
# for the Experimently experimentation platform. Assignment and evaluation are
# decided by the server (X-API-Key authenticated public API); the SDK caches
# the answers per user + key.
#
# Quick start:
#   require 'experimently'
#
#   client = Experimently::Client.new(
#     base_url: "http://localhost:8000",
#     api_key:  ENV["EXPERIMENTLY_API_KEY"]
#   )
#
#   flag = client.evaluate_flag("dark-mode", "user-123")
#   flag.enabled?          # => true or false
#
#   assignment = client.get_assignment("checkout-flow", "user-123")
#   assignment.variant_name if assignment   # => "control" or "treatment"
#
#   client.track("purchase", "user-123", value: 12.5, experiment_key: "checkout-flow")
module Experimently
  # Convenience factory for creating a configured client.
  #
  # @param base_url [String]  API origin, e.g. "http://localhost:8000"
  # @param api_key  [String]  API key
  # @param kwargs   [Hash]    additional SdkConfig options
  # @return [Client]
  def self.configure(base_url:, api_key:, **kwargs)
    Client.new(base_url: base_url, api_key: api_key, **kwargs)
  end
end

File: sdk/flutter/lib/experimently.dart
/// Flutter SDK for Experimently.
///
/// Feature-flag evaluation and A/B experiment assignment are decided by the
/// server (`/api/v1/feature-flags/evaluate/{key}` and
/// `/api/v1/tracking/assign`); results are cached per user + key with a TTL
/// and, optionally, persisted to SharedPreferences for offline fallback.
/// Event tracking never throws.
///
/// ## Getting Started
/// ```dart
/// import 'package:experimently/experimently.dart';
///
/// final client = ExperimentationClient(
///   config: SdkConfig(apiKey: 'your-api-key', baseUrl: 'https://api.example.com'),
///   offlineStore: SharedPreferencesOfflineStore(),
/// );
/// await client.init();
///
/// final flag = await client.evaluateFlag('dark-mode', 'user-123');
/// final assignment = await client.getAssignment('checkout-exp', 'user-123');
/// await client.track('purchase', 'user-123', experimentKey: 'checkout-exp', value: 12.5);
/// await client.close();
/// ```
///
/// For plain Dart (no Flutter) import `experimently_core.dart` instead.
library experimently;

export 'experimently_core.dart';
export 'src/shared_preferences_offline_store.dart';
